packages/menu-surface/index.tsx

```diff
@@ -253,62 +250,60 @@ class MenuSurface extends React.Component<MenuSurfaceProps, MenuSurfaceState> {
     };
 
-    return Object.assign(
-      {
-        addClass: (className: string) => {
-          const classList = new Set(this.state.classList);
-          classList.add(className);
-          this.setState({classList});
-        },
-        removeClass: (className: string) => {
-          const classList = new Set(this.state.classList);
-          classList.delete(className);
-          this.setState({classList});
-        },
-        hasClass: (className: string) => this.classes.split(' ').includes(className),
-        hasAnchor: () => !!this.props.anchorElement,
-        notifyOpen: () => {
-          if (this.registerWindowClickListener) {
-            this.registerWindowClickListener();
-          }
-          this.props.onOpen!();
-        },
-        notifyClose: () => {
-          if (this.deregisterWindowClickListener) {
-            this.deregisterWindowClickListener();
-          }
-          this.props.onClose!();
-        },
-        isElementInContainer: (el: HTMLElement) => {
-          if (!this.menuSurfaceElement.current) return false;
-          if (this.menuSurfaceElement.current === el) {
-            return true;
-          }
-          return this.menuSurfaceElement.current.contains(el);
-        },
-        isRtl: () => {
-          if (!this.menuSurfaceElement) return false;
-          if (!this.menuSurfaceElement.current) return false;
-          return window
-            .getComputedStyle(this.menuSurfaceElement.current)
-            .getPropertyValue('direction') === 'rtl';
-        },
-        setTransformOrigin: (transformOrigin: string) => this.setState({transformOrigin}),
+    return {
+      addClass: (className: string) => {
+        const classList = new Set(this.state.classList);
+        classList.add(className);
+        this.setState({classList});
       },
-      focusAdapterMethods,
-      dimensionAdapterMethods
-    );
+      removeClass: (className: string) => {
+        const classList = new Set(this.state.classList);
+        classList.delete(className);
+        this.setState({classList});
+      },
+      hasClass: (className: string) => this.classes.split(' ').includes(className),
+      hasAnchor: () => !!this.props.anchorElement,
+      notifyOpen: () => {
+        if (this.registerWindowClickListener) {
+          this.registerWindowClickListener();
+        }
+        this.props.onOpen!();
+      },
+      notifyClose: () => {
+        if (this.deregisterWindowClickListener) {
+          this.deregisterWindowClickListener();
+        }
+        this.props.onClose!();
+      },
+      isElementInContainer: (el: HTMLElement) => {
+        if (!this.menuSurfaceElement.current) return false;
+        if (this.menuSurfaceElement.current === el) {
+          return true;
+        }
+        return this.menuSurfaceElement.current.contains(el);
+      },
+      isRtl: () => {
+        if (!this.menuSurfaceElement) return false;
```

Comment:
I've seen many places that you're checking if `this.menuSurfaceElement` is null or undefined. Would this every be null or undefined? Can we not use Non-null assertion operator? (`this.menuSurfaceElement!.current`)

Reply:
There is a time when this could be undefined/null, which is before componentDidMount, and after componentWillUnmount. So assuming that it is `non-null` is not accurate.